### ISSUE TITLE
hover shows (unknown) type for assignments involving imported structs' extern decls

### ISSUE THREAD

**Issue by atroquinine** [zigtools/zls#645]
### Zig Version

0.10.0-dev.3952+9e070b653

### Zig Language Server Version

866655169c48fcf1bf28b27dd3721708ba5a8af7

### Steps to Reproduce

Import a file that has a pub extern declaration (`a`).

Assign this declaration (`a`) to a const / var (`b`).

Hover over the right side of the assignment, right over `a`. It correctly shows its type.

Hover over the new declaration (`b`). It incorrectly shows its type as unknown, even though it is highlighted as a function.

The problem does not occur with `c` and `d` or `e`, `f` and `g` in the example.

lib1.zig:
```zig
const lib2 = @import("lib2.zig");
const b = lib2.a;

pub extern fn c() void;
const d = c;

const e = struct {
    pub extern fn f() void;
};

const g = e.f;
```

lib2.zig:
```zig
pub extern fn a() void;
```

### Expected Behavior

`b` should correctly have its type inferred from the assignment. It seems to almost do that because it is properly highlighted as a function.

### Actual Behavior

`b` shows its type as (unknown) on hover.

**Comment by sqrtroot** [comment]
Any update on this bug, maybe some pointers as to where to look?

**Comment by atroquinine** [comment]
I have not looked into it too much since reporting it, sorry

**Comment by Techatrix** [comment]
I just checked your code example and hovering over `b` correctly shows its function type. If I had to guess this may have been fixed by #1273.
